design_docs/config-file_reorganization.md

```diff
@@ -0,0 +1,137 @@
+<h1> Config File Reorganization <br>
+Xylar Asay-Davis<br>
+date: 01-29-2017<br>
+</h1>
+<h2> Summary </h2>
+This document describes various efforts to clean up the structure of the MPAS-Analysis config file.  The idea is to create a template config file that will replace `config.analysis` as well as a number of example config files designed to make use of various MPAS and ACME runs on various machines.  The reorganization should make the analysis easier for users to modify and run.
+
+<h1> Requirements </h1>
+
+<h2> Requirement: a simple way of turning on and off individual analysis modules <br>
+Date last modified: 2017/01/29 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+There should be a simple, intuitive method for turning on and off individual analysis modules (e.g. `ocean/ohc_timeseries`).  This should replace the current approach of having a boolean `generate` flag for each analysis module in a separate config section.  Preferably, there should be an equivalent method for turning on and off analysis modules from the command line that overrides that in the config file.
+
+<h2> Requirement: there should be a simplified template for config files <br>
+Date last modified: 2017/02/01 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+The current example config file should be made into a general template.  Simplifications should be made to the template so that it can more easily and intuitively be modified for several analyses.  Example config files should also be added for analyzing several existing runs on several different machines.
+
+<h2> Requirement: removal of ACME specific config options <br>
+Date last modified: 2017/02/01 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+To the extent possible, ACME-specific config options such as `casename` and `ref_casename_v0` should be generalized in a way that is also appropriate not just ACME runs but also any other runs involving the MPAS components we support.
+
+<h2> Requirement: consistent section and option names <br>
+Date last modified: 2017/02/01 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+A consistent convention of capitalization and underscores should be used throughout the config file.
+
+
+<h1> Design and Implementation </h1>
+
+<h2> Implementation: a simple way of turning on and off individual analysis modules <br>
+Date last modified: 2017/02/02 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+Implementation of the `config.template` file can be found [here](https://github.com/xylar/MPAS-Analysis/blob/5d5f64bde6ecf1d71f375a61783ff30f1654df01/config.template).
+
```

Comment:
Minor comments:

For https://github.com/xylar/MPAS-Analysis/blob/5d5f64bde6ecf1d71f375a61783ff30f1654df01/config.template#L157 -- we have the option to compare against a model, which is commented out.  I think this is ok for now but just want to make sure we keep that option open for the future if possible.

It would be good to get to the bottom of https://github.com/xylar/MPAS-Analysis/blob/5d5f64bde6ecf1d71f375a61783ff30f1654df01/config.template#L299, but this is not essential.

I'd say go for it!